lib/puppet/ssl/host.rb

```diff
@@ -371,27 +382,67 @@ def state
 
   private
 
+  # Ensures that CRL is either on disk, or that CRL checking has been disabled
+  def ensure_crl_if_needed
+    if use_crl? && !Puppet::FileSystem.exist?(crl_path)
+      # The CertificateRevocationList indirector will attempt to download the
+      # CRL from the CA if it does not exist on disk. It will ask host for
+      # its ssl_store again, but expect crl checking to be disabled for that
+      # store.
+      old_crl_usage = @crl_usage
+      @crl_usage = false
+      Puppet.debug _("Disabling certificate revocation checking when fetching the CRL and no CRL is present")
+      if !CertificateRevocationList.indirection.find(CA_NAME)
+        raise Puppet::Error,
+              _("Certificate revocation checking is enabled but a CRL cannot be found; CRL checking will not be performed.")
+      end
+      @crl_usage = old_crl_usage
+    end
+  end
+
+  # @param path [String] Path to CRL Chain
+  # @return [Array<OpenSSL::X509::CRL>] CRLs from chain
+  # @raise [Errno::ENOENT] if file does not exist
+  # @raise [Puppet::Error<OpenSSL::X509::CRLError>] if the CRL chain is malformed
+  def load_crls(path)
+    ending = "-----END X509 CRL-----\n"
```

Comment:
Oh but the `read` below is text mode too, so it will automatically translate from `\r\n` when it's read in... I'm not sure which is better, using `text` mode and relying on it to do magic, or using `binary` mode and trying to handle `\r\n` or `\n`.

Reply:
I'm thinking UTF8 with String#scan will solve this problem, but I'm not 100%. I've left a comment at https://github.com/puppetlabs/puppet/pull/6797#discussion_r185279615 with my reasons, if you sanity check me.